tuplex/test/core/TakeTest.cc

```diff
@@ -0,0 +1,345 @@
+
+        if (new_a % 2 == 0) {
+            intermediate.emplace_back(new_a, r.getInt(2));
+        }
+    }
+
+    return generateReferenceData(intermediate, topLimit, bottomLimit);
+}
+
+TEST_F(TakeTest, takeMapFilterTest) {
+    mt19937 data_seed_gen(56120);
+
+    const std::vector<size_t> test_size{1, 10, 100, 1001, 10001};
+    const std::vector<size_t> limit_values{0, 1, 5, 11, 600, 10000};
+    const std::vector<string> partition_sizes{"256B", "512KB", "1MB"};
+
+    UDF map_udf("lambda a, b, c: ((a + b), c)");
+    UDF filter_udf("lambda a, b: a % 2 == 0");
+
+    for (auto &part_size: partition_sizes) {
+        auto opt = testOptions();
+        opt.set("tuplex.partitionSize", part_size);
+        Context context(opt);
+
+        for (auto data_size: test_size) {
+            for (auto top_limit: limit_values) {
+                for (auto bottom_limit: limit_values) {
+                    std::cout << "testing with partition size:" << part_size << " data size:"
+                              << data_size << " top:" << top_limit << " bottom:" << bottom_limit << std::endl;
+
+                    auto data = generateTestData(data_size, data_seed_gen());
+                    auto ref_data = generateMapFilterReferenceData(data, top_limit, bottom_limit);
+
+                    auto ds = context.parallelize(data).map(map_udf).filter(filter_udf);
+                    auto res = ds.take(top_limit, bottom_limit);
+                    ASSERT_EQ(ref_data.size(), res->rowCount());
+                    for (Row &r: ref_data) {
+                        Row res_row = res->getNextRow();
+                        if (!(res_row == r)) {
+                            ASSERT_EQ(res_row, r);
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+
+TEST_F(TakeTest, collectIdentityTest) {
+    mt19937 data_seed_gen(123454);
+
+    const std::vector<size_t> test_size{1, 10, 100, 1001, 10001};
+    const std::vector<string> partition_sizes{"256B", "512KB", "1MB"};
+
+    for (auto &part_size: partition_sizes) {
+        auto opt = testOptions();
+        opt.set("tuplex.partitionSize", part_size);
+        Context context(opt);
+
+        for (auto data_size: test_size) {
+            auto data = generateTestData(data_size, data_seed_gen());
+            auto res = context.parallelize(data).collect();
+            ASSERT_EQ(data.size(), res->rowCount());
+            for (Row &r: data) {
+                Row res_row = res->getNextRow();
+                if (!(res_row == r)) {
+                    ASSERT_EQ(res_row, r);
+                }
+            }
+        }
+    }
+}
+
+TEST_F(TakeTest, fileInputTest) {
+    const std::vector<size_t> test_size{1, 1001, 50001};
+    const std::vector<size_t> limit_values{0, 1, 600, 10000};
+    const std::vector<string> partition_sizes{"256B", "1MB"};
+    std::vector<std::vector<Row>> expected_outputs;
+
+    if (!boost::filesystem::exists(scratchDir)) {
```

Comment:
scratchdir should be named with the testname

Reply:
this should already be the testname (done in the testsetup)